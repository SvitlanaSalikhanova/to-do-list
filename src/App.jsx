import './App.css';
import React from 'react';
import Header from './components/Header';
import MainController from './controllers/MainController';

function App() {
    return (
        <div className="App">
            <Header />
            <MainController />
        </div>
    );
}

export default App;
