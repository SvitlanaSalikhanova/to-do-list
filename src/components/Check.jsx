/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import CheckCss from '../styles/Check.module.scss';

export default function Check({ item, onChange }) {
    return (
        <div className={CheckCss.container}>
            <div className={CheckCss.round}>
                <input
                    type="checkbox"
                    checked={item.isDone}
                    onChange={onChange}
                    id={`checkbox${item.id}`}
                />
                <label htmlFor={`checkbox${item.id}`} />
            </div>
        </div>
    );
}
