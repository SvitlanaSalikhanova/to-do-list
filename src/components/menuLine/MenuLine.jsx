import React from 'react';
import { SHOW_ACTIVE, SHOW_COMPLETED, SHOW_ALL } from '../../FilterConstatnts';
import MenuLineCss from './MenuLine.module.scss';

export default function MenuLine({
    count, changeFilter, deleteCompleted, filter,
}) {
    return (
        <div className={MenuLineCss.menuLine}>
            <div className={MenuLineCss.count}>
                {count}
                {' '}
                item
                {count !== 1 && 's'}
            </div>

            <div className={MenuLineCss.filters}>
                <button
                    type="button"
                    className={
                        filter === SHOW_ALL
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => changeFilter(SHOW_ALL)}
                >
                    All
                </button>
                <button
                    type="button"
                    className={
                        filter === SHOW_ACTIVE
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => changeFilter(SHOW_ACTIVE)}
                >
                    Active
                </button>
                <button
                    type="button"
                    className={filter === SHOW_COMPLETED
                        ? MenuLineCss.filterStateActive
                        : MenuLineCss.filterState}
                    onClick={() => changeFilter(SHOW_COMPLETED)}
                >
                    Completed
                </button>
            </div>
            <button
                type="button"
                className={MenuLineCss.clearCompleted}
                onClick={deleteCompleted}
            >
                Clear Completed
            </button>

        </div>
    );
}
