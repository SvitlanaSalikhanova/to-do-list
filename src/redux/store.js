import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './TasksSlice';
import filterReducer from './FilterSlice';

const store = configureStore({
    reducer: {
        tasks: tasksReducer,
        filter: filterReducer,
    },
});

export default store;
