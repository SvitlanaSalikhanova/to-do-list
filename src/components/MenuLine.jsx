import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setFilter } from '../redux/FilterSlice';
import { clearCompleted } from '../redux/TasksSlice';
import { SHOW_ACTIVE, SHOW_COMPLETED, SHOW_ALL } from '../FilterConstatnts';
import MenuLineCss from '../styles/MenuLine.module.scss';

export default function MenuLine({ count }) {
    const filter = useSelector((state) => state.filter.filter);
    const dispatch = useDispatch();

    return (
        <div className={MenuLineCss.menuLine}>
            <div className={MenuLineCss.count}>
                {count}
                {' '}
                item
                {count !== 1 && 's'}
            </div>

            <div className={MenuLineCss.filters}>
                <button
                    type="button"
                    className={
                        filter === SHOW_ALL
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => dispatch(setFilter(SHOW_ALL))}
                >
                    All
                </button>
                <button
                    type="button"
                    className={
                        filter === SHOW_ACTIVE
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => dispatch(setFilter(SHOW_ACTIVE))}
                >
                    Active
                </button>
                <button
                    type="button"
                    className={filter === SHOW_COMPLETED
                        ? MenuLineCss.filterStateActive
                        : MenuLineCss.filterState}
                    onClick={() => dispatch(setFilter(SHOW_COMPLETED))}
                >
                    Completed
                </button>
            </div>
            <button
                type="button"
                className={MenuLineCss.clearCompleted}
                onClick={() => dispatch(clearCompleted())}
            >
                Clear Completed
            </button>

        </div>
    );
}
