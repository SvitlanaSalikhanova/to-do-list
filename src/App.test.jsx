import React from 'react';
import { render as testingRender, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from './redux/store';
import App from './App';
import MenuLine from './components/MenuLine';
import Check from './components/Check';

function render(component) {
    return testingRender(
        <Provider store={store}>
            {component}
        </Provider>,
    );
}

test('renders Todo text', () => {
    render(<App />);
    const linkElement = screen.getByText(/TODO/i);
    expect(linkElement).toBeInTheDocument();
});

const testItemDone = {
    id: 1,
    text: 'Test item 1',
    isDone: true,
};
const testItemNotDone = {
    id: 2,
    text: 'Test item 2',
    isDone: false,
};

describe('Check icon test', () => {
    it.each([[testItemDone, 'BsCheckCircle'], [testItemNotDone, 'BsCircle']])('renders correct check icon', (item, ckeckClassName) => {
        const { container } = render(<Check item={item} />);
        const checkIcons = container.getElementsByClassName(ckeckClassName);
        expect(checkIcons.length).toBe(1);
    });
});

test('renders check marked if task is done', () => {
    const { container } = render(<Check item={testItemDone} />);
    const checkIcon = container.getElementsByClassName('BsCheckCircle');
    expect(checkIcon.length).toBe(1);
});

test('renders menuLine with count', () => {
    render(<MenuLine count={2} />);
    const countElement = screen.getByText(/2 items/i);
    expect(countElement).toBeInTheDocument();
});

describe('MenuLine filters', () => {
    it.each(['All', 'Active', 'Completed'])('renders menuLine with filter buttons', (text) => {
        render(<MenuLine count={2} />);
        const filterElement = screen.getByText(text);
        expect(filterElement).toBeInTheDocument();
    });
});
