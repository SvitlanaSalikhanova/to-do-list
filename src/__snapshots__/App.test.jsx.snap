// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests App Matches Snapshot 1`] = `null`;
